diesel/src/migrations/mod.rs

```diff
@@ -227,12 +230,65 @@ fn search_for_migrations_directory(path: &Path) -> Result<PathBuf, MigrationErro
     }
 }
 
+/// Looks for a migrations directory in the current path and all parent paths,
+/// and creates one in the same directory as the Cargo.toml if it can't find
+/// one. It also sticks a .gitkeep in the directory so git will pick it up.
+/// Returns a `SetupError::CargoTomlNotFound` if no Cargo.toml is found.
+pub fn create_migrations_directory() -> Result<PathBuf, SetupError> {
+    let project_root = try!(find_project_root());
+    try!(fs::create_dir(project_root.join("migrations")));
+    try!(fs::File::create(project_root.join("migrations/.gitkeep")));
+    Ok(project_root)
+}
+
+fn find_project_root() -> Result<PathBuf, SetupError> {
+    search_for_cargo_toml_directory(&try!(env::current_dir()))
+}
+
+fn search_for_cargo_toml_directory(path: &Path) -> Result<PathBuf, SetupError> {
+    let toml_path = path.join("Cargo.toml");
+    if toml_path.is_file() {
+        Ok(path.to_owned())
+    } else {
+        path.parent().map(search_for_cargo_toml_directory)
+            .unwrap_or(Err(SetupError::CargoTomlNotFound))
+    }
+}
+
+/// Creates the database specified in the connection url. It panics if it
+/// can't connect to the postgres connection url, and returns a
+/// `SetupError::QueryError` if it is unable to create the database.
+pub fn create_database(database_url: String) -> Result<(), SetupError> {
+    let mut split: Vec<&str> = database_url.split("/").collect();
+    let database = split.pop().unwrap();
+    let postgres_url = split.join("/");
+    let connection = Connection::establish(&postgres_url)
+        .expect(&format!("Error connecting to {}", postgres_url));
+    try!(connection.execute(&format!("CREATE DATABASE {};", database)));
+    Ok(())
+}
+
+/// Returns true if the '__diesel_schema_migrations' table exists in the
+/// database we connect to, returns false if it does not.
+pub fn schema_table_exists(connection: &Connection) -> bool {
+    let result = connection.execute("SELECT 1
+        FROM information_schema.tables
+        WHERE table_name = '__diesel_schema_migrations';").unwrap();
+    match result {
```

Comment:
Actually it doesn't matter because this will just go away once I redo my metadata stuff because this just becomes `get_metadata_version(&connection) == 0`